HOWTO-make-a-release.md

```diff
@@ -0,0 +1,377 @@
+Check the commits that were added, using for example `git log`.  Check the
+signed announcement .asc file.  Check that the tarball length and hashes
+match in the .md5, .sha1, .sha256, and review the announcment file.
+
+*Do not push* changes to the public repo at this stage.
+(the public repo being `openssl-git@git.openssl.org:openssl.git`)
+
+*Do* send the auto-generated commits to the reviewer and await their
+approval.
+
+### OpenSSL 3.0 and on
+
+The release generating script is in the OpenSSL source checkout, and is
+generally called like this:
+
+    dev/release.sh --reviewer=NAME
+
+This script has a multitude of other options that are useful for specific
+cases, and is also self-documented:
+
+-   To get a quick usage reminder:
+
+        dev/release.sh --help
+
+-   To get a man-page:
+
+        dev/release.sh --manual
+
+### OpenSSL before 3.0
+
+The release generating script is in the tools checkout, represented here
+with $TOOLS, and is generally called like this:
+
+    $TOOLS/release-tools/mkrelease.pl --reviewer=NAME
+
+The manual for that script is found in `$TOOLS/release-tools/MKRELEASE.md`
+
+## Update the website locally
+
+*The changes in this section should be made in your clone of the openssl
+web repo*
+
+Update the news/newsflash.txt file.  This normally is one or two lines.
+Just copy and paste existing announcements making minor changes for the date
+and version number as necessary.  If there is an advisory then ensure you
+include a link to it.
+
+Update the news/vulnerabilities.xml file if appropriate.
+
+If there is a Security Advisory then copy it into the news/secadv directory.
+
+Commit your changes, but *do not push* them to the website repo at this stage.
+(the public repo being `openssl-git@git.openssl.org:openssl-web.git`)
+
+*Do* send the commits to the reviewer and await their approval.
+
+# Publish the release
+
+*BE CAREFUL*  This section makes everything visible and is therefore largely
+irreversible.  If you are performing a dry run then DO NOT perform any steps
+in this section.
+
+Check that the release has been uploaded properly.  The release tarballs and
+associated files should be in ~openssl/dist/new.  They should be owned by
+the openssl userid and world-readable.
+
+Copy the tarballs to appropriate directories.  This can be done using the
+do-release.pl script.  See $TOOLS/release-tools/DO-RELEASE.md for a
+description of the options.  For example:
+
+    sudo -u openssl perl ~openssl/do-release.pl --copy --move
+
+This will copy the relevant files to the website and move them from
+`~openssl/dist/new` to `~openssl/dist/old` so they will not seen by a
+subsequent release.  Alternatively if you want to perform one release at a
+time or copy/move the files manually, see below.
+
+The do-release.pl script will display the commands you will need to issue to
+send the announcement emails later.  Keep a note of those commands for
+future reference.
```

Comment:
I noticed that the commands it tells you to execute don't include the "sudo -u openssl" bit which is inconsistent with these instructions which always include it.

Reply:
At the time, most instructions recommended doing `sudo -u openssl -H bash` and run the commands in that shell.  That's how I usually do stuff.  The do-release.pl script is entirely separate (and in another repo, as it should), so won't be fixed in this PR.

Reply:
Fair enough. I just noticed the inconsistency and thought I'd mention it.